projects/wvr-elements/src/lib/shared/utility/projection.utility.ts

```diff
@@ -13,19 +13,22 @@ const projectSourceContent = (elementRef: ElementRef, sourceElementRef: ElementR
 const project = (element: Element, sourceElementRef: ElementRef, templateSelector: string): void => {
   const templates: Array<HTMLTemplateElement> = Array.from(sourceElementRef.nativeElement.querySelectorAll(templateSelector));
   if (!!element) {
-    templates.forEach((template: HTMLTemplateElement) => {
-      const clone: Node = template.children.length === 0 && template.content.children.length > 0
-        ? template.content.cloneNode(true)
-        : template;
-      Array.from((clone as Element).children)
-        .forEach((elem: Element) => {
-          element.appendChild(elem);
-        });
-    });
-  }
-  // hide target element if nothing to project
-  if (!!element && !templates.length) {
-    (element as HTMLElement).hidden = true;
+    if (templates.length) {
+      templates.forEach((template: HTMLTemplateElement) => {
+        const clone: Node = template.children.length === 0 && template.content.children.length > 0
+          ? template.content.cloneNode(true)
+          : template;
+
+        Array.from((clone as Node).childNodes)
+          .forEach((child: Node) => {
+            element.appendChild(child);
+          });
+      });
+    }
+    else {
+      // hide target element if nothing to project
+      (element as HTMLElement).hidden = true;
```

Comment:
What is the performance difference with this change?

Reply:
An almost insubstantial reduction in one operation.

The old code has: `if (!!element) {` then followed by: `if (!!element && !templates.length) {`.
The new code has: `if (!!element) {` then followed by: `if (templates.length) {`.

This reduces 9 comparison operations to 4 by simply changing code structure:
1.The `element` from the first if block.
2. The `!element`  from the first if block.
3. The `!!element`  from the first if block.
4. The `element` from the second if block.
5. The `!element` from the second if block.
6. The `!!element` from the second if block.
7. The `&&` from the second if block.
8. The `templates.length` from the second if block.
9. The `!templates.length` from the second if block.

Which is reduced to the following:
This is a reduction in of comparison/boolean operations:
1.The `element`  from the first if block.
2. The `!element`  from the first if block.
3. The `!!element`  from the first if block.
4. The `templates.length` from the second if block.

In javascript, this level of performance is rarely payed attention to, but as it didn't hurt the readability of the code, I figured it was worth squeezing out a little more performance.

Reply:
So like fraction of a nanosecond or something like that. :) Is also possible transpilation will optimize the logical branches out into JavaScript. Most compilers would do this and thus readability is preferred. Not saying old or new way is more readable.